### ISSUE TITLE
Add yamllint

### ISSUE THREAD

**Issue by luciecastella** [IAMconsortium/nomenclature#176]
Following @phackstock 's suggestion, we should inclue the YAML linter that checks for syntax validity, indentation, trailing whitespaces,... (see if checks for whitespaces at the end of codes and hidden characters are still useful or not)

**Comment by phackstock** [comment]
This is the package in question that I found: https://pypi.org/project/yamllint/. Might be able to replace/enhance some of the yaml checking we are currently doing.

**Comment by luciecastella** [comment]
I checked more closely the package and how it works. It seems to be not totally relevant for our use-case: 

- It checks whitespaces but only at the end of a line since it takes a yaml file as input: I cannot see own to use that for codes, and the trainling whitespace on a line is actually not a problem except for aestetics
- I ran it with the special character, and it does not find it. 
- One thing that is maybe interesting is that it can check for key duplicates, but only if keys are listed in the form:

> \- key1
> key2
> key 3

and not
> \- key1
> \- key2
> \- key 3

which is currently used... 

- Another thing would be the check for octal values: if a value is given with a 0 at the beginning, yaml interprets it as octal, and this check therefore avoid any further error based on this behaviour. not sure it is useful though

- It also checks if there are too many whitespaces, if brackets are empty, if a blank line is present at the end and basically aestetics things that would certainly make the upload fail more often for details...

So basically me question is: For what feature did you originally think this would be useful @phackstock ? Maybe we could drop the idea of using it? @danielhuppmann

**Comment by danielhuppmann** [comment]
Agree, doesn't seem very useful... Thanks for investigating @luciecastella!
